com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.hlsl

```diff
@@ -391,73 +391,91 @@ void LightLoop( float3 V, PositionInputs posInput, PreLightData preLightData, BS
-            builtinData.backBakeDiffuseLighting = (uninitializedGI && !apvEnabled) ? float3(0.0, 0.0, 0.0) : builtinData.backBakeDiffuseLighting;
-        }
+            BuiltinData tempBuiltinData;
+            ZERO_INITIALIZE(BuiltinData, tempBuiltinData);
 
-        if (apvEnabled)
-        {
-            if (uninitializedGI)
+#if !defined(_SURFACE_TYPE_TRANSPARENT)
+            if (_IndirectDiffuseMode != INDIRECTDIFFUSEMODE_OFF)
             {
-                float3 R = reflect(-V, bsdfData.normalWS);
-                // Need to make sure not to apply ModifyBakedDiffuseLighting() twice to our bakeDiffuseLighting data, which could happen if we are dealing with initialized data (light maps).
-                // Create a local BuiltinData variable here, and then add results to builtinData.bakeDiffuseLighting at the end.
-                BuiltinData apvBuiltinData;
-                ZERO_INITIALIZE(BuiltinData, apvBuiltinData);
-                SetAsUninitializedGI(apvBuiltinData.bakeDiffuseLighting);
-                SetAsUninitializedGI(apvBuiltinData.backBakeDiffuseLighting);
-
-                EvaluateAdaptiveProbeVolume(GetAbsolutePositionWS(posInput.positionWS),
-                    bsdfData.normalWS,
-                    -bsdfData.normalWS,
-                    R,
-                    V,
-                    posInput.positionSS,
-                    apvBuiltinData.bakeDiffuseLighting,
-                    apvBuiltinData.backBakeDiffuseLighting,
-                    lightInReflDir);
-
-                float indirectDiffuseMultiplier = GetIndirectDiffuseMultiplier(builtinData.renderingLayers);
-                apvBuiltinData.bakeDiffuseLighting *= indirectDiffuseMultiplier;
-                apvBuiltinData.backBakeDiffuseLighting *= indirectDiffuseMultiplier;
+                tempBuiltinData.bakeDiffuseLighting = LOAD_TEXTURE2D_X(_IndirectDiffuseTexture, posInput.positionSS).xyz * GetInverseCurrentExposureMultiplier();
+            }
+            else
+#endif
+            {
+#if defined(PROBE_VOLUMES_L1) || defined(PROBE_VOLUMES_L2)
+                if (_EnableProbeVolumes)
+                {
+                    // Reflect normal to get lighting for reflection probe tinting
+                    float3 R = reflect(-V, bsdfData.normalWS);
+
+                    EvaluateAdaptiveProbeVolume(GetAbsolutePositionWS(posInput.positionWS),
+                        bsdfData.normalWS,
+                        -bsdfData.normalWS,
+                        R,
+                        V,
+                        posInput.positionSS,
+                        tempBuiltinData.bakeDiffuseLighting,
+                        tempBuiltinData.backBakeDiffuseLighting,
+                        lightInReflDir);
+                }
+                else // If probe volume is disabled we fallback on the ambient probes
+                {
+                    tempBuiltinData.bakeDiffuseLighting = EvaluateAmbientProbe(bsdfData.normalWS);
```

Comment:
@FrancescoC-unity I change the code of APV which handle the case of disable the APV. Now nothing is done in Material Pass and all is done in Lightloop. This was a requirement to be compliant with the new system but it is also cleaner.